TPI/juegoDeLaVida/solucion.cpp

```diff
@@ -25,13 +34,50 @@ vector<posicion> posicionesVivas(toroide t) {
 
 /***************************** EJERCICIO densidadPoblacion ******************************/
 float densidadPoblacion(toroide t) {
-    float densidad = 0.0;
-    return densidad;
+
+    if (!esValido(t))
+        return 0.0;
+
+    int cantVivas = posicionesVivas(t).size();
+    int cantTotal = filas(t) * columnas(t);
+
+    return cantVivas / cantTotal;
 }
 
+
 /**************************** EJERCICIO evolucionDePosicion *****************************/
+bool estaViva(toroide t, int f, int c){
+    return t[f][c];
+}
+
+int vivasAdyacentes(toroide t, int f, int c){
+    int count = 0;
+
+    for (int i = -1; i <= 1; i++) {
+        for (int j = -1; j <= 1; j++){
+            posicion p = make_tuple(f + i, c + j);
+            if (valorPosicionNormalizada(t, p))
```

Comment:
Si hacemos esto, podemos sacarnos de encima el `if (estaViva)` de más abajo. Básicamente, si `j == 0 && i == 0` no contamos la posición
```suggestion
            if ((i != 0 || j != 0) && valorPosicionNormalizada(t, p))
```